### ISSUE TITLE
PuckAnimator animates with `duration=0` when called with an empty `options` lambda.

### ISSUE THREAD

**Issue by tomaszrybakiewicz** [mapbox/mapbox-maps-android#1446]
**tl;dr;** 
Android SDK (31) `ValueAnimator.clone()` method doesn't copy `duration` value, causing `PuckAnimator.userConfiguredAnimator.duration = 0`.

## Environment

- Android OS version: `Android 12 (API 31)`
- Devices affected: `Pixel 6`
- Maps SDK Version: `10.6.0`

## Observed behavior and steps to reproduce
We detected this issue when working on a patch for a [Location puck's velocity](https://github.com/mapbox/mapbox-navigation-android/issues/4140).

[Puck Velocity Fix](https://github.com/mapbox/mapbox-navigation-android/pull/5925) relies on a mechanism that allows customization of ValueAnimator via `LocationConsumer.onLocationUpdated()` callback.

When `NavigationLocationProvider` calls `LocationConsumer.onLocationUpdated(vararg location: Point, options: (ValueAnimator.() -> Unit)? = null)` with _non-null_ `options`, the `PuckAnimator` uses a separate animator instance  (`userConfiguredAnimator`) to drive the animation (`PuckAnimator.animate()`).

```kotlin 
// PuckAnimator.kt
class PuckAnimator(...) : ValueAnimator() {
  ... 
  init {
    setObjectValues(emptyArray<Any>())
    setEvaluator(evaluator)
    addUpdateListener {
      @Suppress("UNCHECKED_CAST")
      val updatedValue = it.animatedValue as T
      updateLayer(it.animatedFraction, updatedValue)
      updateListener?.invoke(updatedValue)
    }
    duration = LocationComponentConstants.DEFAULT_INTERVAL_MILLIS
    interpolator = DEFAULT_INTERPOLATOR
    userConfiguredAnimator = clone() // <---------- THIS doesn't copy duration value
  }
  ...
  fun animate(
    vararg targets: T,
    options: (ValueAnimator.() -> Unit)? = null
  ) {
    cancelRunning()
    if (options == null) {
      setObjectValues(*targets)
      start()
    } else {  // <------------ 
      options.invoke(userConfiguredAnimator)
      userConfiguredAnimator.setObjectValues(*targets)
      userConfiguredAnimator.start()
    }
  }
  ...
}
```

The `userConfiguredAnimator` instance is created via `ValueAnimator.clone()` method in the `PuckAnimator` constructor. It appears that cloned version doesn't copy duration field, leaving `userConfiguredAnimator.duration = 0`.

(screenshots of debbuger session showing PuckAnimator duration fields)
| | |
|:---:|:---:|
|<img width="1552" alt="Screen Shot 2022-06-17 at 3 51 07 PM" src="https://user-images.githubusercontent.com/2678039/174395066-a250d581-fd3b-4cde-9142-6fb83daf0b22.png"/>|<img width="1552" alt="Screen Shot 2022-06-17 at 3 51 18 PM" src="https://user-images.githubusercontent.com/2678039/174395067-2cbe36ab-1378-40e9-91c8-b72b90caaa00.png"/>|

<details><summary>Expand for glitch recording</summary>

https://user-images.githubusercontent.com/2678039/174207047-6719eeb9-5d77-422f-bcb3-529ae8bb6cfc.mp4

</details>

## Expected behaviour

Calling `PuckPositionAnimator.animate()` with an empty `options` block should animate Puck position with default duration (`LocationComponentConstants.DEFAULT_INTERVAL_MILLIS`).

Both animate calls below should result in the same animator behaviour:

```kotlin
var positionAnimator = PuckPositionAnimator(...)
val point = Point.fromLngLat(..., ...)

// First call without options
positionAnimator.animate(point)    

// Second call with empty options block
positionAnimator.animate(point, options = {})
```

## Notes / preliminary analysis

After further investigation, we identified that in Android 12 SDK (API 31) neither the [ValueAnimator.clone()](https://android.googlesource.com/platform/frameworks/base/+/android12-release/core/java/android/animation/ValueAnimator.java#1572), not its super class [Animator.clone()](https://android.googlesource.com/platform/frameworks/base/+/android12-release/core/java/android/animation/Animator.java#398)  clones the duration value, leaving it `0`.

## Proposed solution

PuckAnimator should set `userConfiguredAnimator.duration` default value to match `PuckAnimator.duration`.

```kotlin
// PuckAnimator.kt
class PuckAnimator(...) : ValueAnimator() {
  ... 
  init {
    setObjectValues(emptyArray<Any>())
    setEvaluator(evaluator)
    addUpdateListener {
      @Suppress("UNCHECKED_CAST")
      val updatedValue = it.animatedValue as T
      updateLayer(it.animatedFraction, updatedValue)
      updateListener?.invoke(updatedValue)
    }
    duration = LocationComponentConstants.DEFAULT_INTERVAL_MILLIS
    interpolator = DEFAULT_INTERPOLATOR
    userConfiguredAnimator = clone() // <---------- THIS doesn't copy duration value
    userConfiguredAnimator.duration = duration  // <---- MANUALLY copy duration value
  }
...
}
```

**Comment by tomaszrybakiewicz** [comment]
@kiryldz, Thanks. This issue is not a blocker and we already [implemented a workaround in NavSDK](https://github.com/mapbox/mapbox-navigation-android/pull/5943/files#diff-f2cca061380015a1e107803f17368bee5d2cfd1575726972f387b2549232fa5cR174). However this issue might still affect developers that decide to use their LocationProvider implementation with LocationComponentPlugin, instead suggested NavigationLocationProvider.
